components/datadog/agent/docker.go

```diff
@@ -133,6 +133,11 @@ func defaultAgentParams(params *dockeragentparams.Params) {
 			p.FullImagePath = fmt.Sprintf("%s-jmx", p.FullImagePath)
 		}
 	}(params)
+	defer func(p *dockeragentparams.Params) {
+		if p.FIPS {
+			p.FullImagePath += "-fips"
+		}
+	}(params)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
How does this work if both jmx and fips are set?

Reply:
What about having them in a single defer function? Might help with readability